solver_spec.rb

```diff
@@ -0,0 +1,41 @@
+require_relative 'solver'
+
+RSpec.describe Solver do
+  let(:solver) { Solver.new }
+
+  describe '#factorial' do
+    it 'Returns 1 when the inputed number is 0' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'Returns the factorial of a positive integer' do
+      expect(solver.factorial(8)).to eq(40_320)
+      expect(solver.factorial(4)).to eq(24)
+    end
```

Comment:
- [ ] As required for the project, kindly ensure that the `#factorial` method also has tests which handle `Negative value which raises an exception as an expectation.`